### PR TITLE
add INetworkClient

### PR DESCRIPTION
**Pull request by uweenukr**

### CODE REVIEW

**Comment by uweenukr** [comment]
I have similar branches for INetworkServer and INetworkManager. If this is accepted I will open PRs for those also.

File: Assets/Mirror/Runtime/INetworkClient.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Mirror
+{
+    public interface ClientObjectsManager
```
Comment (paulpach) [inline Assets/Mirror/Runtime/INetworkClient.cs:7]:
```suggestion
    public interface IClientObjectsManager
```

File: Assets/Mirror/Runtime/INetworkClient.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        void RegisterSpawnHandler(Guid assetId, SpawnHandlerDelegate spawnHandler, UnSpawnDelegate unspawnHandler);
+
+        void UnregisterSpawnHandler(Guid assetId);
+
+        void ClearSpawners();
+
+        void DestroyAllClientObjects();
+    }
+
+    public interface ClientSceneManager
```
Comment (paulpach) [inline Assets/Mirror/Runtime/INetworkClient.cs:34]:
```suggestion
    public interface IClientSceneManager
```

File: Assets/Mirror/Runtime/INetworkClient.cs (hunk 1 above)
Comment (uweenukr) [reply Assets/Mirror/Runtime/INetworkClient.cs:7]: omg im dum lol

**Pull request merged**